### DISCUSSION TITLE
Custom dataset with edge features

### DISCUSSION THREAD

**Discussion by marco-fuchs** [pyg-team/pytorch_geometric#6064] [Q&A] [answered]
Hi, I am relatively new to PyTorch geometric. 
Basically I need a graph with heterogeneous edge features. I know there are various gnn-layers that support this edge features but haven't found a holistic example of how these can be read from a CSV into e.g. HeteroData? Are there any examples or tutorials?

Thanks for your help!
Marco

**Comment by wsad1** [comment, marked as answer]
https://pytorch-geometric.readthedocs.io/en/latest/notes/load_csv.html, maybe this will help.

**Reply by marco-fuchs** [reply]
That helps - thanks a lot
